src/structures/InodeMap.cpp

```diff
@@ -0,0 +1,60 @@
+#include "../includes/InodeMap.hpp"
+
+// Get nice inode hash
+int InodeMap::get_inode_hash(str_t name) {
+    return std::hash<str_t>{} (name);   // TODO: use our own hash func
+}
+
+Inode InodeMap::get_inode(str_t src_name){
+    try {
+        int magic_number = get_inode_hash(src_name);
+        return inode_map.at(magic_number);      
+    } catch (std::out_of_range) { throw InodeMapException("There's no inode with name " + src_name); }
+};
+
+// Add inode to InodeMap, write to FS bin file
+void InodeMap::add_inode(bool src_type, str_t src_name) {
+    int magic_number = get_inode_hash(src_name);
+    if(inode_map.find(magic_number) != inode_map.end()) {
+        throw InodeMapException("Impossible to add Inode with name '" + src_name + "' because it already exists");
+    }
+    inode_map[magic_number] = Inode(src_type);
+};
+
+void InodeMap::add_inode(bool src_type, str_t src_name, int address_block) {
+    int magic_number = get_inode_hash(src_name);
+    if(inode_map.find(magic_number) != inode_map.end()) {
+        throw InodeMapException("Impossible to add Inode with name '" + src_name + "' because it already exists");
+    }
+    inode_map[magic_number] = Inode(src_type, address_block);
+};
```

Comment:
Maybe function `add_inode` should receive an Inode, not its fields? Idea - is to store src name in Inode and then transmit it everywhere we need. In that case there will be no need to implement different kinds of same function.